javascript/README.md

```diff
@@ -88,6 +88,21 @@ var length = items.length;
 var name = 'foo';
 ````
 
+## Don't have unused variables/expressions
+```javascript
+// Bad: Many of these variables/expressions are useless
+{} // object - not used
+var newArray = []; // newArray - not used
+['1', '2', '3', '4'].forEach(function(num, index) { // index - not used
```

Comment:
What about when something you need is in the second argument? Say we wanted index but not num - a functional programming common practice is to use `_` in it's place, but will the linter complain about that? If so I don't think we should add it to the linting rules.